route/build/cpl/RtmMod.F90

```diff
@@ -429,19 +448,29 @@ SUBROUTINE route_run(rstwr)
       if(ierr/=0)then; call shr_sys_abort(trim(subname)//trim(cmessage)); endif
     enddo
 
+    call t_stopf('mizuRoute_subcycling')
+
+    !-----------------------------------
     ! getting ready for exporting
+    !-----------------------------------
+    call t_startf('mizuRoute_prep_export')
+
+    rtmCTL%discharge = 0._r8
+    rtmCTL%volr      = 0._r8
+    rtmCTL%flood     = 0._r8
+
     ! put reach flux variables to associated HRUs
     if (multiProcs) then
       if (masterproc) then
         nRch_trib=rch_per_proc(0)
-        if (nRch_mainstem/=0) then
+        if (nRch_mainstem > 0) then
```

Comment:
I'm glad you are switching the exact match for an inequality which is more robust.